spec/rspec/matchers/built_in/raise_error_spec.rb

```diff
@@ -290,13 +290,28 @@ def to_s
   end
 end
 
-RSpec.describe "expect { ... }.not_to raise_error(message)" do
-  it "issues a warning" do
-    expect_warning_with_call_site __FILE__, __LINE__+1, /risks false positives/
+RSpec.describe "expect { ... }.not_to raise_error('message')" do
+  it "issues a warning when configured to do so", :warn_about_potential_false_positives do
+    RSpec::Expectations.configuration.warn_about_potential_false_positives = true
+    expect_warning_with_call_site __FILE__, __LINE__+1, /not_to raise_error\(message\)` risks false positives/
+    expect { raise 'blarg' }.not_to raise_error('blah')
+  end
+
+  it "supresses the warning when configured to do so", :warn_about_potential_false_positives do
+    RSpec::Expectations.configuration.warn_about_potential_false_positives = false
+    expect_no_warnings
+    expect { raise 'blarg' }.not_to raise_error('blah')
+  end
+end
+
+RSpec.describe "expect { ... }.not_to raise_error(/message/)" do
+  it "issues a warning when configured to do so", :warn_about_potential_false_positives do
+    RSpec::Expectations.configuration.warn_about_potential_false_positives = true
+    expect_warning_with_call_site __FILE__, __LINE__+1, /not_to raise_error\(message\)` risks false positives/
```

Comment:
Just to dissect it. Before the change, this was printing a slightly different warning:
```ruby
Using `expect { }.not_to raise_error(SpecificErrorClass)` risks false positives
```
and now it's a more appropriate for `not_to raise_error(/regexp/)`:
```ruby
Using `expect { }.not_to raise_error(message)` risks false positives
```

My apologies that it took that long to figure out.